src/inmanta/resources.py

```diff
@@ -239,7 +239,7 @@ def add_reference(self, path: str, reference: "references.Reference[references.P
         )
 
 
-def collect_references(value_reference_collector: ReferenceCollector | None, value: object, path: str) -> object:
+def collect_references(value_reference_collector: ReferenceCollector, value: object, path: str) -> object:
```

Comment:
this the None gone, I wonder if this should be a method on the ReferenceCollector?